leases/src/main/java/com/redhat/ipaas/leases/impl/SqlLeaseManager.java

```diff
@@ -0,0 +1,283 @@
+            dbi.update("CREATE TABLE IF NOT EXISTS leases (lease VARCHAR PRIMARY KEY, owner VARCHAR, expiration BIGINT)");
+        });
+    }
+
+    public void dropTables() {
+        withDB(dbi -> {
+            dbi.update("DROP TABLE leases");
+        });
+    }
+
+    public SqlLease create(String leaseId, Consumer<Lease> consumer) {
+        SqlLease l = new SqlLease(this, leaseId, consumer);
+        // synchronize this cause we want to make sure the user does not create duplicate leases.
+        synchronized (this) {
+            if (leases.containsKey(leaseId)) {
+                throw new IllegalStateException("A lease already exists for: " + leaseId);
+            }
+            leases.put(leaseId, l);
+        }
+        executor.execute(this::doLeaseMaintenance);
+        return l;
+    }
+
+    @Override
+    public String getOwnerId() {
+        return ownerId;
+    }
+
+    public static class LeaseOwner {
+        private String lease;
+        private String owner;
+
+        public String getLease() {
+            return lease;
+        }
+
+        public void setLease(String leaseId) {
+            this.lease = leaseId;
+        }
+
+        public String getOwner() {
+            return owner;
+        }
+
+        public void setOwner(String owner) {
+            this.owner = owner;
+        }
+    }
+
+    @Override
+    public Map<String, String> getAllLeases() {
+        HashMap<String, String> result = new HashMap<>();
+        withDB(dbi -> {
+            dbi.createQuery("SELECT lease,owner FROM leases")
+                    .map(LeaseOwner.class).forEach(x -> {
+                result.put(x.getLease(), x.getOwner());
+            });
+        });
+        return result;
+    }
+
+    protected void release(SqlLease lease) {
+        synchronized (this) {
+            leases.remove(lease.getLeaseId());
+        }
+        executor.execute(() -> {
+            if (lease.isAquired()) {
+                lease.fireReleased();
+                withDB(dbi -> {
+                    dbi.createStatement("DELETE FROM leases where lease = :lease and owner = :owner")
+                            .bind("lease", lease.getLeaseId())
+                            .bind("owner", ownerId)
+                            .execute();
+                });
+            }
+        });
+    }
+
+    protected void refresh() {
+        executor.execute(() -> {
+            doLeaseMaintenance();
+            try {
+                scheduler.schedule(this::refresh, leaseDuration / 2, TimeUnit.MILLISECONDS);
+            } catch (RejectedExecutionException ignore) {
+            }
+        });
+    }
+
+    private void doLeaseMaintenance() {
+        withDB(dbi -> {
+            long now = currentTimeMillis();
+
+            // Delete expired leases..
+            dbi.createStatement("DELETE FROM leases where :now > expiration")
+                    .bind("now", now)
+                    .execute();
+
+            // Refresh our leases.
```

Comment:
I think each one of those SQL statements can be executed concurrently and multiple times without issue.  No transactions are needed. BONUS: less locking. 

Can you think of scenario where thats not the case?